queries.sql

```diff
@@ -14,3 +14,17 @@ SELECT * FROM animals WHERE name != 'Gabumon';
 
```

Comment:
Please create queries to answer the following questions in transactions:

Please UPDATE the animals table by setting the species column to unspecified, then roll back the change. For
example:
  `BEGIN; -- start transaction
   UPDATE animals
   SET species = 'unspecified'; -- make change
   SELECT species from animals; -- verify that change was made
   ROLLBACK;
   SELECT species from animals; -- verify that change was undone`

Please UPDATE the animals table by setting the species column to digimon for all animals that have a name ending in mon.
please UPDATE the animals table by setting the species column to pokemon for all animals that don't have species already set. For example:
`  BEGIN;
   UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
   UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 
   SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit
`
Please delete all records in the animals table, then roll back the transaction. For example:
 ` BEGIN;
   DELETE FROM animals WHERE ;
   SELECT COUNT(*) FROM ANIMALS;
   ROLLBACK;
   SELECT COUNT(*) FROM ANIMALS;
`
Please DELETE all animals born after Jan 1st, 2022
please Create a savepoint for the transaction
Please UPDATE all animals' weight to be their weight multiplied by -1.
Then Rollback to the savepoint
Please UPDATE all animals' weights that are negative to be their weight multiplied by -1 and then Commit the transaction. For example 
  ` BEGIN;
    DELETE FROM animals WHERE  date_of_birth >= '2022/01/01' ;
    SAVEPOINT SP1;
    UPDATE animals SET weight_kg = weight_kg * -1; 
    ROLLBACK TO SP21;
    UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0; 
    COMMIT;
`

Reply:
Hi @SajjadAhmad14 ,
Thank you for your feedback,
I've done all of these transactions, but in the requirements of the project they are not required to be reported in any files of the repo.[Project requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md)